sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -98,20 +101,38 @@ internal class StyleObserver(
         styleDataStyleLoadedListener = null
       }
       StyleDataType.SPRITE -> {
-        styleDataSpritesLoadedListener?.onStyleLoaded(
-          loadedStyle ?: throw MapboxMapException("Style is not initialized yet although SPRITES event has arrived!")
-        )
-        styleDataSpritesLoadedListener = null
+        onStyleSpritesReady()
       }
       StyleDataType.SOURCES -> {
-        styleDataSourcesLoadedListener?.onStyleLoaded(
-          loadedStyle ?: throw MapboxMapException("Style is not initialized yet although SOURCES event has arrived!")
-        )
-        styleDataSourcesLoadedListener = null
+        onStyleSourcesReady()
       }
     }
   }
 
+  private fun onStyleSourcesReady() {
+    // sources listener is not null - it means style does not contain them thus StyleDataType.SOURCES
+    // was not fired, explicitly call it to bind all the necessary style extensions upstream
+    styleDataSourcesLoadedListener?.let {
+      // reset listener first, firing onStyleLoaded
+      // may produce another StyleDataType.SOURCES event if sources are added
+      styleDataSourcesLoadedListener = null
+      it.onStyleLoaded(
+        loadedStyle ?: throw MapboxMapException("Style is not initialized yet although SOURCES event has arrived!")
```

Comment:
Can you verify whether this works as expected if switch two styles very quickly?

Reply:
I've tested it in the app loading 10+ styles, but let me try to do the instrumentation test